datafusion_iceberg/src/table.rs

```diff
@@ -478,7 +478,15 @@ async fn table_scan(
             .collect()
     });
 
-    // Create plan for every parition with delete files
+    let file_source = if let Some(physical_predicate) = physical_predicate.clone() {
+        Arc::new(
+            ParquetSource::default().with_predicate(Arc::clone(&file_schema), physical_predicate),
+        )
+    } else {
+        Arc::new(ParquetSource::default())
+    };
```

Comment:
Please double check if this is the proper way of creating this file_source.

Reply:
From what I understand the ParquetSource looks good.

Now that I'm looking into it, what do you think about setting [with_pushdown_filters](https://docs.rs/datafusion/latest/datafusion/datasource/physical_plan/parquet/source/struct.ParquetSource.html#method.with_pushdown_filters) to true. It's false by default.

It should improve the performance.

Reply:
Added to it